nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/main/java/org/apache/nifi/web/security/request/ContentLengthFilter.java

```diff
@@ -0,0 +1,143 @@
+
+/**
+ * This {@link Filter} rejects HTTP requests that exceed a specific, maximum size.
+ */
+public class ContentLengthFilter implements Filter {
+    private static final Logger logger = LoggerFactory.getLogger(ContentLengthFilter.class);
+    public final static String MAX_LENGTH_INIT_PARAM = "maxContentLength";
+    public final static int MAX_LENGTH_DEFAULT = 10_000_000;
+    private int maxContentLength;
+
+    public ContentLengthFilter() {
+        maxContentLength = MAX_LENGTH_DEFAULT;
+    }
+
+    public ContentLengthFilter(int maxLength) {
+        maxContentLength = maxLength;
+    }
+
+    @Override
+    public void init(FilterConfig config) throws ServletException {
+        String maxLength = config.getInitParameter(MAX_LENGTH_INIT_PARAM);
+        int length = maxLength == null ? MAX_LENGTH_DEFAULT : Integer.parseInt(maxLength);
+        if (length < 0) {
+            throw new ServletException("Invalid max request length.");
+        }
+        maxContentLength = length;
+        logger.info("Max content length set: " + maxLength + "b");
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain next) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        String httpMethod = httpRequest.getMethod();
+
+        // Check the HTTP method because the spec says clients don't have to send a content-length header for methods
+        // that don't use it.  So even though an attacker may provide a large body in a GET request, the body should go
+        // unread and a size filter is unneeded at best.  See RFC 2616 section 14.13, and RFC 1945 section 10.4.
+        boolean willReadInputStream = maxContentLength > 0 && (httpMethod.equalsIgnoreCase("POST") || httpMethod.equalsIgnoreCase("PUT"));
+        if (!willReadInputStream) {
+            logger.info("No length check of request with method {} and maximum {}", httpMethod, maxContentLength);
+            next.doFilter(request, response);
+            return;
+        }
+
+        HttpServletResponse httpResponse = (HttpServletResponse) response;
+        int contentLength = request.getContentLength();
+        if (contentLength > maxContentLength) {
+            // Request with a client-specified length greater than our max is rejected:
+            logger.info("Content length check rejected request with content-length {} greater than maximum {}", contentLength, maxContentLength);
+            httpResponse.setContentType("text/plain");
+            httpResponse.getOutputStream().write("Payload Too large".getBytes());
+            httpResponse.setStatus(HttpServletResponse.SC_REQUEST_ENTITY_TOO_LARGE);
+        } else {
+            // If or when the request is read, this limits the read to our max:
+            logger.info("Content length check allowed request with content-length {} less than maximum {}", contentLength, maxContentLength);
```

Comment:
I think this should be set to `debug` to avoid flooding the logs.

Reply:
Changed to `debug` and added units.